### ISSUE TITLE
Class ordering in parallel execution doesn't work

### ISSUE THREAD

**Issue by stykalin** [junit-team/junit5#3399]
I enabled parallel execution and added `@Order` annotations to all classes for sorting.
I implemented `CustomClassOrdering` for checking class sorting and as I can see in log files (`target/.logs/testSorting_before.txt` and `target/.logs/testSorting_after.txt`) it's working.

But when the tests run, the test class with the lowest order value runs at the same time as the test class with the highest order value. This can be seen in the execution window in the IntellijI DEA and in a log file (`target/.logs/testsByThread.txt`)

Please check the repository to reproduce https://github.com/stykalin/junit5-parallel-ordering

## Context

 - Used versions (Jupiter/Vintage/Platform):  junit-jupiter 5.9.3
 - Build Tool/IDE: IntellijI DEA 2023.1.4, Maven

## Deliverables
![image](https://github.com/junit-team/junit5/assets/18271575/cdeaf9af-75d3-4c92-af0b-1ca5ac9f7bd6)
![image](https://github.com/junit-team/junit5/assets/18271575/3508f520-e095-4f97-86eb-2160fda11817)
![image](https://github.com/junit-team/junit5/assets/18271575/52c10dc7-671b-4cb0-a864-d3d284ae6687)

**Comment by dhoard** [comment]
Appears to be working as designed. See https://github.com/junit-team/junit5/issues/3007 (specifically https://github.com/junit-team/junit5/issues/3007#issuecomment-1482686556)

**Comment by marcphilipp** [comment]
Duplicate of #3007
